lib/standalone-single-spa.js

```diff
@@ -58,98 +58,103 @@ class StandaloneSingleSpaPlugin {
     compiler.hooks.compilation.tap(pluginName, (compilation) => {
       let publicPath;
 
-      this.options.HtmlWebpackPlugin.getHooks(
-        compilation
-      ).beforeAssetTagGeneration.tap(pluginName, (data) => {
-        publicPath = data.assets.publicPath;
-      });
+      const oldHtmlPlugin = !this.options.HtmlWebpackPlugin.getHooks;
 
-      this.options.HtmlWebpackPlugin.getHooks(compilation).alterAssetTags.tap(
-        pluginName,
-        (data) => {
-          const mainScript = data.assetTags.scripts[0];
-          const mainScriptSrc = publicPath || "/" + mainScript.attributes.src;
+      if (oldHtmlPlugin) {
+        compilation.plugin(
+          "html-webpack-plugin-alter-asset-tags",
+          (data, cb) => {
+            this.modifyScripts({ publicPath, scripts: data.body });
```

Comment:
I didn't find a way to get the publicPath in html-webpack-plugin@3. So I just left it blank, knowing that if people use a custom webpack public path they'll probably open an issue to get this fixed.